--- passgen.py	(original)
+++ passgen.py	(refactored)
@@ -1,11 +1,11 @@
 #passgen 0.3.5#
 import sys, random, string, subprocess
 
-print '''
+print('''
 .---..---..---..---..---..---..-..-.
 | |-'| | | \ \  \ \ | |'_| |- | .` |
 `-'  `-^-'`---'`---'`-'-/`---'`-'`-'
-               0.3.7'''
+               0.3.7''')
 
 len = 8
 
@@ -14,7 +14,7 @@
 		try:
 			char_set = 'abcdefghijklmnopqrstuvwxyz234567'
 			result = ''.join(random.sample(char_set*6, len))
-			print result
+			print(result)
 		except (KeyboardInterrupt):
 			exit()
 def hexdigits():
@@ -22,7 +22,7 @@
 		try:
 			char_set = string.hexdigits
 			result = ''.join(random.sample(char_set*6, len))
-			print result
+			print(result)
 		except (KeyboardInterrupt):
 			exit()
 def lowercase():
@@ -30,7 +30,7 @@
 		try:
 			char_set = string.ascii_lowercase
 			result = ''.join(random.sample(char_set*6, len))
-			print result
+			print(result)
 		except (KeyboardInterrupt):
 			exit()
 def lowerupper():
@@ -38,7 +38,7 @@
 		try:
 			char_set = string.ascii_lowercase + string.ascii_uppercase
 			result = ''.join(random.sample(char_set*6, len))
-			print result
+			print(result)
 		except (KeyboardInterrupt):
 			exit()
 def lowernum():
@@ -46,7 +46,7 @@
 		try:
 			char_set = string.ascii_lowercase + string.digits
 			result = ''.join(random.sample(char_set*6, len))
-			print result
+			print(result)
 		except (KeyboardInterrupt):
 			exit()
 def uppernum():
@@ -54,7 +54,7 @@
 		try:
 			char_set = string.ascii_uppercase + string.digits
 			result = ''.join(random.sample(char_set*6, len))
-			print result
+			print(result)
 		except (KeyboardInterrupt):
 			exit()
 def loweruppernum():
@@ -62,7 +62,7 @@
 		try:
 			char_set = string.ascii_uppercase + string.ascii_lowercase + string.digits
 			result = ''.join(random.sample(char_set*6, len))
-			print result
+			print(result)
 		except (KeyboardInterrupt):
 			exit()
 def uppercase():
@@ -70,21 +70,21 @@
 		try:
 			char_set = string.ascii_uppercase
 			result = ''.join(random.sample(char_set*6, len))
-			print result
+			print(result)
 		except (KeyboardInterrupt):
 			exit()
 def aircrack():
 	arglist()
-	characterset = raw_input('Enter permutation set: ')
-	bssid = raw_input('Enter bssid: ')
-	capfile = raw_input('Enter capfile: ')
+	characterset = input('Enter permutation set: ')
+	bssid = input('Enter bssid: ')
+	capfile = input('Enter capfile: ')
 	try:
 		cmd = (['python passgen.py ' + characterset + ' | sudo aircrack-ng --bssid ' + bssid + ' -w- ' + capfile])
 		proc = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True, shell=True)
 		while proc.poll() == None:
 			pcrackOut = proc.stdout
 			nextline = proc.stdout.readlines()
-			print nextline
+			print(nextline)
 		exit()
 	except KeyboardInterrupt:
 		proc.terminate()
@@ -92,7 +92,7 @@
 		return
 		exit()
 	except Exception as e:
-		print e
+		print(e)
 		proc.terminate()
 		proc.wait()
 		exit()
@@ -128,12 +128,12 @@
 					char_set = sys.argv[2]
 					char_len = sys.argv[3]
 					result = ''.join([random.choice(char_set) for _ in range(int(char_len))])
-					print result
+					print(result)
 				except (IndexError):
-					print(IndexError, "Make sure you've added your character map and length")
+					print((IndexError, "Make sure you've added your character map and length"))
 					exit() 
 				except (ValueError):
-					print(ValueError, 'ValueError: sample larger than population')
+					print((ValueError, 'ValueError: sample larger than population'))
 					exit()
 				except (KeyboardInterrupt):
 					exit()
